src/eos-dropbox-app

```diff
@@ -183,35 +187,39 @@ class DropboxLauncher():
-                                           'org.freedesktop.portal.OpenURI',
-                                           None)
+            self._bus = Gio.bus_get_sync(Gio.BusType.SESSION, None)
+            self._proxy = Gio.DBusProxy.new_sync(self._bus, Gio.DBusProxyFlags.NONE,
+                                                 None,
+                                                 'org.freedesktop.portal.Desktop',
+                                                 '/org/freedesktop/portal/desktop',
+                                                 'org.freedesktop.portal.OpenURI',
+                                                 None)
 
             logging.info("Opening Dropbox directory at {}...".format(path))
-            handle = proxy.call_with_unix_fd_list('OpenFile',
-                                                  GLib.Variant('(sha{sv})',
-                                                               ('', 0, GLib.Variant('a{sv}', None))),
-                                                  Gio.DBusCallFlags.NONE,
-                                                  -1,
-                                                  Gio.UnixFDList.new_from_array([fd]))
-
-            # The OpenURI method returns a handle to a 'request object', which stays
-            # alive for the duration of the user interaction related to the method call,
-            # so we connect to its Response signal to know when it's all over.
-            bus.signal_subscribe('org.freedesktop.portal.Desktop',
-                                 'org.freedesktop.portal.Request',
-                                 'Response',
-                                 handle,
-                                 None,
-                                 Gio.DBusSignalFlags.NO_MATCH_RULE,
-                                 self._responseReceived)
-
-        except GLib.Error as e:
-            self._exitOnError("Could not launch the OpenURI portal for {}: {}".format(uri, e.message))
+            [result, _] = self._proxy.call_with_unix_fd_list_sync('OpenFile',
+                                                                  GLib.Variant('(sha{sv})',
+                                                                               ('', 0, None)),
+                                                                  Gio.DBusCallFlags.NONE,
+                                                                  -1,
+                                                                  Gio.UnixFDList.new_from_array([fd]),
+                                                                  None)
+            handle = result.unpack()[0];
+            self._bus.signal_subscribe('org.freedesktop.portal.Desktop',
+                                       'org.freedesktop.portal.Request',
+                                       'Response',
+                                       handle,
+                                       None,
+                                       Gio.DBusSignalFlags.NO_MATCH_RULE,
```

Comment:
Wowee, the portal uses directed signals? Cool!

This code to subscribe to the signal is duplicated, verbatim, in `_fallbackToOpenUriMethod`. I think you should return `handle` from that method, or even inline it in the `except` block, apart from the signal binding code it's a one-linerr.

Reply:
> Wowee, the portal uses directed signals? Cool!

Yes it does: https://github.com/flatpak/xdg-desktop-portal/blob/d3cd1119422120e486dee25cecff81c78eeab17d/src/request.c#L51

Reply:
I'll refactor the code around the directed signals not to duplicate, ok